src/Source.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Spatie\ResponsiveImages;
+
+use Exception;
+use Illuminate\Contracts\Support\Arrayable;
+use Illuminate\Support\Collection;
+use Spatie\ResponsiveImages\Jobs\GenerateImageJob;
+
+/**
+ * Representing <source> tag in HTML
+ * @property-read Breakpoint $breakpoint
+ */
+class Source implements Arrayable
+{
+    public Breakpoint $breakpoint;
+    protected string $format;
+    protected string $mediaWidthUnit;
+
+    public function __construct(Breakpoint $breakpoint, ?string $format = 'original')
+    {
+        $this->breakpoint = $breakpoint;
+        $this->format = $format;
+        $this->mediaWidthUnit = config('statamic.responsive-images.breakpoint_unit', 'px');
+    }
+
+    public function __set($name, $value): void
+    {
+        throw new Exception(sprintf('Cannot modify property %s', $name));
+    }
+
+    public function getMimeType(): string|null
+    {
+        $mimeTypesBySetFormat = [
+            'webp' => 'image/webp',
+            'avif' => 'image/avif',
+        ];
+
+        if (isset($mimeTypesBySetFormat[$this->format])) {
+            return $mimeTypesBySetFormat[$this->format];
+        }
+
+        return $this->breakpoint->asset->mimeType();
+    }
```

Comment:
This actually benefits for the `original` format, as previously the `<source>` tag for it did not contain mimetype at all, and browser has more difficult time without a mimetype specified. For example, if source asset was AVIF image, then `original` srcset could be unnecessarily loaded by the browser and may make the image broken, as AVIF image may not be supported by some browsers.  I don't think browsers can guess mimetype from the provided URLs that Glide URL generator gives.